shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -168,37 +197,36 @@ - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent*)event {
 }
 
 - (void)applyBlurBackdropFilters:(NSMutableArray<PlatformViewFilter*>*)filters {
-  BOOL needUpdateFilterViews = NO;
-  if (self.filters.count != filters.count) {
-    needUpdateFilterViews = YES;
-  } else {
-    for (NSUInteger i = 0; i < filters.count; i++) {
-      if (!CGRectEqualToRect(self.filters[i].frame, filters[i].frame) ||
-          !flutter::BlurRadiusEqualToBlurRadius(self.filters[i].blurRadius,
-                                                filters[i].blurRadius)) {
-        needUpdateFilterViews = YES;
-      }
-    }
+  FML_DCHECK(self.filters.count == self.backdropFilterSubviews.count);
+  if (self.filters.count == 0 && filters.count == 0) {
+    return;
   }
-  if (needUpdateFilterViews) {
-    // Clear the old filter views.
-    for (PlatformViewFilter* filter in self.filters) {
-      [[filter backdropFilterView] removeFromSuperview];
-    }
-    // Update to the new filters.
-    self.filters = [filters retain];
-    // Add new filter views.
-    for (PlatformViewFilter* filter in self.filters) {
-      UIView* backdropFilterView = [filter backdropFilterView];
+  self.filters = filters;
+  NSUInteger index = 0;
+  for (index = 0; index < self.filters.count; index++) {
+    UIVisualEffectView* backdropFilterView;
+    PlatformViewFilter* filter = self.filters[index];
+    if (self.backdropFilterSubviews.count <= index) {
+      backdropFilterView = filter.backdropFilterView;
       [self addSubview:backdropFilterView];
+      [self.backdropFilterSubviews addObject:backdropFilterView];
+    } else {
+      [filter updateVisualEffectView:self.backdropFilterSubviews[index]];
```

Comment:
Would it be possible to just update the CAFilters instead so you don't have to remove the existing view from its superview?

Reply:
This is what the code is doing: update the CAFilters and Frame of the existing views.